evalml/tests/component_tests/test_arima_regressor.py

```diff
@@ -34,348 +34,247 @@ def test_model_instance(ts_data):
     assert isinstance(fitted, ARIMARegressor)
 
 
-def test_get_dates_fit_and_predict(ts_data):
+def test_fit_ts_without_y(ts_data):
     X, y = ts_data
-    clf = ARIMARegressor()
-    date_col, X_ = clf._get_dates(X, y)
-    assert isinstance(date_col, pd.DatetimeIndex)
-    assert X_.equals(X)
-
 
-def test_match_indices(ts_data):
-    X, y = ts_data
-    date_index = pd.date_range("2020-10-02", "2020-11-01")
     clf = ARIMARegressor()
-    X_, y_ = clf._match_indices(X, y, date_index)
-    assert isinstance(X_.index, pd.DatetimeIndex)
-    assert isinstance(y_.index, pd.DatetimeIndex)
-    assert X_.index.equals(y_.index)
-    assert X_.index.equals(date_index)
-
+    with pytest.raises(ValueError, match="ARIMA Regressor requires y as input."):
+        clf.fit(X=X)
 
-@pytest.mark.parametrize("predict", [True, False])
-@pytest.mark.parametrize("dates_shape", [0, 1, 2])
-def test_format_dates(predict, dates_shape, ts_data):
-    X, y = ts_data
-    date_index = pd.date_range("2020-10-02", "2020-11-01")
-    if dates_shape == 1:
-        date_index = pd.DataFrame(date_index)
-    elif dates_shape == 2:
-        date_index = pd.DataFrame(data={"a": date_index, "b": date_index})
 
-    clf = ARIMARegressor()
+@pytest.fixture
+def get_X_y():
+    def _get_X_y(
+        train_features_index_dt,
+        train_target_index_dt,
+        train_none,
+        datetime_feature,
+        no_features,
+        test_features_index_dt,
+    ):
+        X = pd.DataFrame(index=[i + 1 for i in range(50)])
+        dates = pd.date_range("1/1/21", periods=50)
+        feature = [i for i in range(50)]
+        y = pd.Series([1, 2, 3, 4, 5, 6, 5, 4, 3, 2] * 5)
+
+        X_train = X[:40]
+        X_test = X[40:]
+        y_train = y[:40]
+
+        if train_features_index_dt:
+            X_train.index = dates[:40]
+        if train_target_index_dt:
+            y_train.index = dates[:40]
+        if test_features_index_dt:
+            X_test.index = dates[40:]
+        if not no_features:
+            X_train["Feature"] = feature[:40]
+            X_test["Feature"] = feature[40:]
+            if datetime_feature:
+                X_train["Dates"] = dates[:40]
+                X_test["Dates"] = dates[40:]
+        if train_none:
```

Comment:
The difference between `train_none` and `no_features` is that `train_none` passes in None as X, while `no_features` passes in a dataframe with only an index set.